logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/e2ei/CheckCrlWorker.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ * Wire
+ * Copyright (C) 2024 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.feature.e2ei
+
+import com.wire.kalium.logic.data.e2ei.CrlRepository
+import com.wire.kalium.logic.data.sync.IncrementalSyncRepository
+import com.wire.kalium.logic.data.sync.IncrementalSyncStatus
+import com.wire.kalium.logic.feature.e2ei.usecase.CheckRevocationListUseCase
+import com.wire.kalium.logic.functional.map
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.filter
+import kotlinx.datetime.Clock
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.hours
+
+/**
+ * Represents an interface for a CheckCrlWorker,
+ * which is should keep the [CrlRepository] healthy.
+ *
+ * It will wait until the incremental sync is live and then check the CRLs if needed.
+ */
+internal interface CheckCrlWorker {
```

Comment:
Suggestion: Can we add here maybe the expanded acronym of `CRL` ? Since this is the use case and most top surface for clients, we can give some lead to devs or more meaning to CRL.

Personally, I don't know what is this :)

Reply:
Certificate Revocation List